### PR TITLE
update typeaheads

### PR DESCRIPTION
**Pull request by KatherynRizek**
Short description explaining the high-level reason for the pull request
Update typeaheads to use `react-bootstrap-typeahead` 

## Additions

- Add AsyncTypeahead, ClearButton, Input, and Typeahead components
- Add react bootstrap typeahead package and proxy

## Changes
- Update CompanyTypeahead, Issue, FederalState, ZipCode   to be functional components, use new typeaheads, 
- Update Typeahead.less to make react bootstrap typeaheads to match current typeaheads
- Update filters and search bar cypress tests to work with the new typeaheads

## Testing

Try out each of the typeahead filters and make sure they are keyboard accessible: 
- Issue
- State
- ZIP code
- Company name
Try out the main Searchbar. There should be a typeahead for the Company search within setting but not for All data or Narratives 

## Screenshots
Screen Shot 2022-11-07 at 5 38 11 PM
Screen Shot 2022-11-07 at 5 38 26 PM
Screen Shot 2022-11-07 at 5 38 45 PM
Screen Shot 2022-11-07 at 5 39 01 PM

## Todos
- Move new functional components created in this PR to be in their own folder with their test file

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,9 +93,10 @@
     "prettier": "^2.7.1",
     "prop-types": "^15.5.10",
     "query-string": "^5.0.0",
-    "react": "16.14.0",
-    "react-dom": "16.14.0",
-    "react-intl": "6.0.5",
+    "react": "^18.2.0",
```
Comment (flacoman91) [inline package.json:96]: 😄

File: cypress/e2e/common/filters.cy.js [hunk 2]
```diff
@@ -237,8 +237,9 @@ describe('Filter Panel', () => {
       cy.log('open again');
       cy.get('.state .o-expandable_link').click();
       cy.log('searches a typeahead filter');
-      cy.get('.state input').clear();
-      cy.get('.state input').type('texas');
+      cy.findByPlaceholderText('Enter state name or abbreviation')
```
Comment (flacoman91) [inline cypress/e2e/common/filters.cy.js:240]: i love these RTL selectors!

File: src/components/Filters/__tests__/Company.spec.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ function setupSnapshot(initialFixture) {
 }
 
 describe('component::Company', () => {
-  describe('snapshots', () => {
+  xdescribe('snapshots', () => {
```
Comment (flacoman91) [inline src/components/Filters/__tests__/Company.spec.js:42]: does this test need to be updated like Issue?

Comment (flacoman91) [reply src/components/Filters/__tests__/Company.spec.js:42]: never mind.  I see the component still needs to be converted to functional.

File: cypress/e2e/common/filters.cy.js (hunk 2 above)
Comment (KatherynRizek) [reply cypress/e2e/common/filters.cy.js:240]: Me too! They've been helping out with a lot of timeout issues

File: cypress/e2e/common/search-bar.cy.js [hunk 4]
```diff
@@ -40,30 +38,51 @@ describe('Search Bar', () => {
 
   describe('Typeaheads', () => {
     it('has no typeahead functionality in All Data', () => {
-      cy.get(searchInput).clear();
-      cy.get(searchInput).type('bank', { delay: 200 });
-      cy.get('@typeahead.all').should('have.length', 0);
+      cy.intercept(typeAheadRequest, { body: [] }).as('typeahead');
+      cy.findByPlaceholderText('Enter your search term(s)').clear();
+      cy.findByPlaceholderText('Enter your search term(s)').type('bank', {
```
Comment (flacoman91) [inline cypress/e2e/common/search-bar.cy.js:43]: can we chain these off of the clear call?

File: src/components/Filters/Issue.spec.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import { defaultAggs } from '../../reducers/aggs/aggs';
+import { defaultQuery } from '../../reducers/query/query';
+import { Issue } from './Issue';
+import { listOfIssues } from '../../testUtils/aggsConstants';
+import * as filterActions from '../../actions/filter';
+
+describe('Issue', () => {
+  const user = userEvent.setup({ delay: null });
+  const renderComponent = () => {
+    const newAggsState = { issue: listOfIssues };
```
Comment (flacoman91) [inline src/components/Filters/Issue.spec.js:18]: 👍

File: src/components/Search/SearchBar.js [hunk 6]
```diff
@@ -1,231 +1,166 @@
...
-      // sync local state from redux
-      this._updateLocalState(searchField, searchText);
+export const SearchBar = ({ debounceWait }) => {
+  const dispatch = useDispatch();
+  const searchField = useSelector(selectQuerySearchField);
+  const searchText = useSelector(selectQuerySearchText);
+  const hasAdvancedSearchTips = useSelector(selectViewHasAdvancedSearchTips);
+  const [inputValue, setInputValue] = useState(searchText);
+  const [dropdownOptions, setDropdownOptions] = useState([]);
+  // handleClear is called whenever the user submits by pressing enter
```
Comment (flacoman91) [inline src/components/Search/SearchBar.js:31]: Thanks for these comments

**Review by flacoman91 — APPROVED** [review]
Looks good.  I tested out the functionality and ran the unit tests and they pass.

I just added some feedback on a few minor changes you could fix if you want. Not required.

The updated keyboard accessibility will be a welcome bonus.

File: cypress/e2e/common/search-bar.cy.js (hunk 4 above)
Comment (KatherynRizek) [reply cypress/e2e/common/search-bar.cy.js]: Good catch. Done in 69ffde1

**Comment by KatherynRizek** [comment]
> Everything looks really good. Thank you. it's a lot of work that went into this.
> 
> One minor nit, can you add to the style for the typeahead pagination-option
> 
> .typeahead .dropdown-item.rbt-menu-pagination-option
> 
> `width:auto;`
> 
> When the "Display additional results..." thing appears in the dropdowns, it's adding a horizontal scrollbar.
> 
> Screen Shot 2022-11-14 at 11 12 43 AM
> 
> Screen Shot 2022-11-14 at 11 12 50 AM

Done in 69ffde1. I also fixed the overflow so there wasn't a need to scroll vertically either

**Pull request merged**